src/main/java/com/hermesworld/ais/galapagos/subscriptions/service/impl/SubscriptionServiceImpl.java

```diff
@@ -80,6 +77,15 @@ public CompletableFuture<SubscriptionMetadata> addSubscription(String environmen
                     .failedFuture(new IllegalArgumentException("Cannot subscribe to application internal topics"));
         }
 
+        List<SubscriptionMetadata> subscriptionsForTopic = this.getSubscriptionsForTopic(environmentId,
+                subscriptionMetadata.getTopicName(), true);
+        for (var subscription : subscriptionsForTopic) {
+            if (Objects.equals(subscription.getClientApplicationId(), subscriptionMetadata.getClientApplicationId())) {
+                return CompletableFuture
+                        .failedFuture(new IllegalArgumentException("Cannot subscribe to the same topic twice"));
```

Comment:
Better: "A subscription of this topic for this application already exists." (more the general style of Galapagos error messages)